src/database/connection.ts

```diff
@@ -0,0 +1,55 @@
+import 'reflect-metadata';
+import { createConnection, ConnectionOptions } from 'typeorm';
+import { NotablePerson } from './entities/notablePerson';
+import { NotablePersonEvent } from './entities/event';
+import { NotablePersonLabel } from './entities/label';
+import { NotablePersonEventComment } from './entities/comment';
+import { User } from './entities/user';
+import { readJson } from '../helpers/readFile';
+import { isUsingProductionDatabase } from './env';
+
+const {
+  // These variables are for the development database
+  // provided by AWS Elastic Beanstalk
+  RDS_DB_NAME,
+  RDS_HOSTNAME,
+  RDS_PORT,
+  RDS_USERNAME,
+  RDS_PASSWORD,
+} = process.env;
+
+const getConfig = async (): Promise<ConnectionOptions> => ({
+  type: 'mysql',
+
+  ...isUsingProductionDatabase
+    ? {
+        ...await readJson<DatabaseConfig>('secrets/db.production.json'),
+
+        // Do not set any of these to `true`,
+        // otherwise the database will be destroyed
+        synchronize: false,
+        dropSchema: false,
+      }
+    : {
+        database: RDS_DB_NAME || 'hollowverse',
+        host: RDS_HOSTNAME || 'localhost',
+        port: Number(RDS_PORT) || 3306,
+        username: RDS_USERNAME || 'root',
+        password: RDS_PASSWORD || '123456',
+        synchronize: true,
+        dropSchema: false,
+      },
+});
+
+export const connection = getConfig().then(async config =>
+  createConnection({
+    ...config,
+    entities: [
+      NotablePerson,
+      NotablePersonEvent,
+      NotablePersonLabel,
+      NotablePersonEventComment,
+      User,
+    ],
```

Comment:
They are created when the connection is established if they have not been created before, otherwise they are reused. If the schema is different, TypeORM can automatically synchronize the schema, although they do not recommend that for production databases. If we need to update the schema we would need to write a "migration", which is a way to describe the changes: add columns, set default values for new columns, delete tables... etc. TypeORM supports migrations.